Makefile

```diff
@@ -4,52 +4,88 @@ else
 ENV_PATH_SEP := :
 endif
 
-
 THIS_MAKEFILE_PATH := $(word $(words $(MAKEFILE_LIST)),$(MAKEFILE_LIST))
 THIS_DIR := $(shell cd $(dir $(THIS_MAKEFILE_PATH));pwd)
 
-NPM ?= $(NODE) $(shell which npm)
-NPM_BIN = $(shell npm bin)
-
 RED=`tput setaf 1`
 GREEN=`tput setaf 2`
+CYAN=`tput setaf 6`
 RESET=`tput sgr0`
 
-START_APP := @$(NPM_BIN)/electron .
-ELECTRON_TEST := ELECTRON_PATH=$(NPM_BIN)/electron $(NPM_BIN)/electron-mocha
-CONFIG := $(THIS_DIR)/desktop/config.json
-DESKTOP_CONFIG := $(THIS_DIR)/desktop-config
-BUILDER := $(THIS_DIR)/builder.js
-BUILD_CONFIG := $(THIS_DIR)/resource/build-scripts/build-config-file.js
-BUILD_DIR := $(THIS_DIR)/build
-PACKAGE_DMG := $(THIS_DIR)/resource/build-scripts/package-dmg.js
-PACKAGE_WIN32 := @$(NPM_BIN)/electron-builder
-CERT_SPC := $(THIS_DIR)/resource/secrets/automattic-code.spc
-CERT_PVK := $(THIS_DIR)/resource/secrets/automattic-code.pvk
 CALYPSO_DIR := $(THIS_DIR)/calypso
-CALYPSO_JS := $(CALYPSO_DIR)/public/build.js
-CALYPSO_CHANGES_STD := `find "$(CALYPSO_DIR)" -newer "$(CALYPSO_JS)" \( -name "*.js" -o -name "*.jsx" -o -name "*.json" -o -name "*.scss" \) -type f -print -quit | grep -v .m. | wc -l`
-CALYPSO_BRANCH = $(shell git --git-dir ./calypso/.git branch | sed -n -e 's/^\* \(.*\)/\1/p')
-WEBPACK_BIN := $(NPM_BIN)/webpack
 
-# sets to 1 if NPM version is >= 3
-NPMGTE3 := $(shell expr `npm -v | cut -f1 -d.` \>= 3)
-MAKENSIS_VERSION := $(shell expr `makensis -VERSION 2>/dev/null`)
+CHECKMARK=✓
+
+CONFIG_ENV = 
+CALYPSO_ENV = desktop
+NODE_ENV = production
+BUILD_PLATFORM = mwl
+
+# Build sources
+# TODO: run tasks parallel when in dev mode
+build-source: checks build-config build-calypso build-desktop
+	@echo "$(CYAN)$(CHECKMARK) All parts built$(RESET)"
+
+# Start app
+start: build-source
+	@echo "$(CYAN)Starting app...$(RESET)"
+
+	@npx electron .
+
+# Build config
+build-config: 
+	@node $(THIS_DIR)/resource/build-scripts/build-config-file.js $(CONFIG_ENV)
```

Comment:
at this point there's not much we're using our `Makefile` for since all it's doing is duplicating messaging and calling Javascript files.

```Makefile
desktop-config/config-%.json:
	$(info Config file for environment "$(CONFIG_ENV)" does not exist. Ignoring Environment.)

desktop/config.json: BASE=desktop-config/config-base.json
desktop/config.json: ENV=desktop-config/config-$(CONFIG_ENV).json
desktop/config.json: desktop-config/config-base.json desktop-config/config-$(CONFIG_ENV).json
	@node -e "var base = require('$(BASE)'), env = require('$(ENV)'); console.log( Object.assign( base, env ) )" > $@
```

with this reconstruction `make` will only rebuild the configuration if either the base or the env-specific config has changed, and we don't have to call for it indirectly as `build-config` - we just declare our dependency as it is - a file. the node script remains minimal and while it's a long line we can see what its goal is and easily change it without worrying about reproducing the right behavior